sql/core/src/test/scala/org/apache/spark/sql/DataFrameSuite.scala

```diff
@@ -1658,4 +1658,49 @@ class DataFrameSuite extends QueryTest with SharedSQLContext {
     val df = spark.createDataFrame(spark.sparkContext.makeRDD(rows), schema)
     assert(df.filter($"array1" === $"array2").count() == 1)
   }
+
+  test("SPARK-18207: Compute hash for wider table") {
+    import org.apache.spark.sql.types.{StructType, StringType}
```

Comment:
I think this test is more like an end-to-end test. The hashing is not obvious as we seen.

Can we add unit test? In `HashExpressionsSuite`, I think? So you can test `HiveHash` too.